superset/db_engine_specs.py

```diff
@@ -980,19 +981,82 @@ def get_columns(
             result.append(cls._create_column_info(column, column.Column, column_type))
         return result
 
+    @classmethod
+    def _is_column_name_quoted(cls, column_name: str) -> bool:
+        """
+        Check if column name is in quotes
+        :param column_name: column name
+        :return: boolean
+        """
+        return column_name.startswith('"') and column_name.endswith('"')
+
+    @classmethod
+    def _get_fields(cls, cols: List[dict]) -> List[ColumnClause]:
+        """
+        Format column clauses where names are in quotes and labels are specified
+        :param cols: columns
+        :return: column clauses
+        """
+        column_clauses = []
+        dot_regex = r'\.(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)'
+        for col in cols:
+            # get individual column names
+            col_names = re.split(dot_regex, col['name'])
+            # quote each column name if it is not already quoted
+            for index, col_name in enumerate(col_names):
+                if not cls._is_column_name_quoted(col_name):
+                    col_names[index] = '"{}"'.format(col_name)
+            quoted_col_name = '.'.join(col_names)
```

Comment:
Nit, you can use a generator expression here (and also an f-string):

```python
quoted_col_name = '.'.join(
  col_name if cls._is_column_name_quoted(col_name) else f'"{col_name}"'
  for col_name in col_names
)
```

Reply:
Sweet. Thanks. My python is a bit rusty so this is definitely more pythonic.